Add conflict status to OpenAPI

kperk103 opened the pull request:
This adds conflict HTTP status under `PATCH` and `PUT` operations.
I was unsure how to know the cause of the conflict so the message
for each conflict status says "Conflict with current state." If
there is anything I can look at to know how to make the message
better/more specific please let me know.

Closes #50

Commit:
Add Conflict HTTP Status

This adds conflict HTTP status under `PATCH` and `PUT` operations.
I was unsure how to know the cause of the conflict so the message
for each conflict status says "Conflict with current state." If
there is anything I can look at to know how to make the message
better/more specific please let me know.

tuckerzp commented on openapi.yaml:
```diff
@@ -211,6 +211,9 @@ paths:
           content: {}
+        409:
+          description: Conflict with current state
+          content: {}
```
I feel like using an empty content tag here is not necessary, From what I can tell, it adds a little `media type` tag that does not provide any functionality. I don't think you should remove it everywhere we do this, but I am somewhat against adding that if not necessary.

kperk103 replied:
I agree. I removed the empty content tag from the 409 status codes that I had previously included.

tuckerzp reviewed (changes requested):
I like the use of a `409` error here. I have not looked into what would fit best a ton, but that seems to fit very well. As far as the `description` tag goes, it seems like we never get _too_ specific with the errors. I am leaning towards `Conflict with current state` being good enough,

Commit:
Remove empty content from 409 status code

Removes "content: {}" from 409 status code because it
appears to not provide any functionality.

Bronstrom commented on openapi.yaml:
```diff
@@ -248,6 +250,8 @@ paths:
           description: Bad Request
         415:
           description: Unsupported media type
+        409:
+          description: Conflict with current state
```
A little nitpicky, but I think we should order these error responses in numerical order. It seems like 404 appears before 400 a few times too — while unrelated to 409, I would suggest we swap these in this PR as well.

Bronstrom reviewed (changes requested):
This is looking great so far! I have one suggestion:

Bronstrom commented:
Also, somewhat out of scope of this issue and a small detail, but it seems like our choice to include `content: {}` is ambiguous throughout the file, while some 4XX error responses to `PUT` and `PATCH` operations include this while others don't. Like @tuckerzp in his request, I don't believe these provide any value at all so these could just be removed. Although being unrelated to the 409 error, perhaps this is something that should be addressed in a separate PR?

kperk103 commented:
> Also, somewhat out of scope of this issue and a small detail, but it seems like our choice to include `content: {}` is ambiguous throughout the file, while some 4XX error responses to `PUT` and `PATCH` operations include this while others don't. Like @tuckerzp in his request, I don't believe these provide any value at all so these could just be removed. Although being unrelated to the 409 error, perhaps this is something that should be addressed in a separate PR?

I definitely agree. I can work on that and create a PR for it. The empty content tags are also sometimes in operations other than `PUT` and `PATCH`. Do you recommend I remove those as well for consistency?

Commit:
Order status codes numerically

This will reorder some status codes to make sure
they are in numerical order.

Bronstrom commented:
> I definitely agree. I can work on that and create a PR for it. The empty content tags are also sometimes in operations other than PUT and PATCH. Do you recommend I remove those as well for consistency?

Yeah, that makes more sense. I would just remove them everywhere they appear in the file. Thanks!

tuckerzp reviewed (approved):
Thanks for your work on this!

tuckerzp committed:
Merge branch 'develop' into feature/update-conflict-http-status
